geode-core/src/test/java/org/apache/geode/internal/cache/tier/sockets/ClientUserAuthsTest.java

```diff
@@ -45,37 +47,78 @@ public void before() throws Exception {
   }
 
   @Test
-  public void putSubjectWithNegativeOneWillProduceNewId() throws Exception {
+  public void putSubjectWithNegativeOneWillProduceNewId() {
     Long uniqueId = auth.putSubject(subject1, -1);
-    verify(auth).removeSubject((Subject) null);
+    verify(auth, never()).removeSubject(anyLong());
     assertThat(uniqueId).isEqualTo(123L);
   }
 
   @Test
-  public void putSubjectWithZeroWillProduceNewId() throws Exception {
+  public void putSubjectWithZeroWillProduceNewId() {
     Long uniqueId = auth.putSubject(subject1, 0);
-    verify(auth).removeSubject((Subject) null);
+    verify(auth, never()).removeSubject(anyLong());
     assertThat(uniqueId).isEqualTo(123L);
+    verify(subject1, never()).logout();
   }
 
   @Test
-  public void putSubjectWithExistingId() throws Exception {
+  public void putSubjectWithExistingId() {
     Long uniqueId = auth.putSubject(subject1, 456L);
-    verify(auth).removeSubject((Subject) null);
+    verify(auth, never()).removeSubject(anyLong());
     assertThat(uniqueId).isEqualTo(456L);
+    verify(subject1, never()).logout();
   }
 
   @Test
-  public void replacedSubjectShouldLogout() throws Exception {
+  public void replacedSubjectShouldNotLogout() {
     Long id1 = auth.putSubject(subject1, -1);
     Long id2 = auth.putSubject(subject2, id1);
     assertThat(id1).isEqualTo(id2);
-    verify(auth).removeSubject(eq(subject1));
+    verify(auth, never()).removeSubject(anyLong());
+    verify(subject1, never()).logout();
+
+  }
+
+  @Test
+  public void getSubjectReturnsTheLatestOne() {
+    Long id1 = auth.putSubject(subject1, -1);
+    assertThat(auth.getSubject(id1)).isSameAs(subject1);
+    auth.putSubject(subject2, id1);
+    assertThat(auth.getSubject(id1)).isSameAs(subject2);
+  }
+
+  @Test
+  public void getSubjectOfNonExistentId() {
```

Comment:
Could this test name include the expected behaviour? Something like "getSubjectWithNonExistentIdReturnsNull" maybe? This also applies to `removeSubjectOfNonExistentId()`, `getSubjectWithCq()` and `removeSubject()`.